backend/src/auth/repositories/auth_repo.py

```diff
@@ -1,32 +1,62 @@
-from fastapi import HTTPException
+import logging
+import uuid
+from datetime import timedelta
+from typing import Optional
+
+from fastapi import Response
+from sqlalchemy.exc import SQLAlchemyError
 from sqlalchemy.orm import Session
 
-from src.auth.interfaces.repository import Repository
+from src.auth.interfaces import Repository
 from src.auth.models import User
-from src.auth.schemas import SignInSchema, UserSchema
+from src.auth.schemas import (
+    RegisterRequest,
+    RegisterResponse,
+    SignInRequest,
+    SignInResponse,
+    UserResponse,
+)
 from src.auth.utils import check_password, hash_password, return_token
 
+logger = logging.getLogger(__name__)
+
 
 class AuthRepository(Repository):
-    def register(self, user: UserSchema, db: Session) -> dict:
+    def get_user_by_id(self, user_id: uuid.UUID, db: Session) -> Optional[UserResponse]:
+        try:
+            user = db.query(User).filter_by(id=user_id).first()
+            return user
+        except Exception as e:
+            logger.error(f"Failed to get user: {e}")
+
+    def get_user(self, username: str, db: Session) -> Optional[UserResponse]:
+        try:
+            user = db.query(User).filter_by(username=username).first()
+            return user
+        except Exception as e:
+            logger.error(f"Failed to get user: {e}")
```

Comment:
This is wrong. You are not returning UserResponse schema, you are returing User from db instance. Change this typing.

Reply:
> This is wrong. You are not returning UserResponse schema, you are returing User from db instance. Change this typing.

Ok, databases return model, views return schema.

Reply:
No, persistance layer return db orm instances, in further layer you will convert it to your schema and then it will return instance of schema